shipyard-controller/api/project.go

```diff
@@ -307,12 +307,21 @@ func (pm *projectManager) updateProject(params *operations.CreateProjectParams)
 		}
 	}
 
-	pm.projectAPI.UpdateConfigurationServiceProject(keptnapimodels.Project{
+	_, errObj := pm.projectAPI.UpdateConfigurationServiceProject(keptnapimodels.Project{
 		GitRemoteURI: params.GitRemoteURL,
 		GitToken:     params.GitToken,
 		GitUser:      params.GitUser,
 		ProjectName:  *params.Name,
 	})
+
+	if errObj != nil {
+		msg := fmt.Sprintf("Could not update upstream repository of project %s: %s", *params.Name, *errObj.Message)
+
+		if delErr := pm.deleteUpstreamRepoCredentials(params); delErr != nil {
```

Comment:
If I'm not mistaken, this will only delete the upstream repo credentials, but not restore potentially previously correct credentials.

Reply:
true. Although that raises to question if we do currently actually support the use case of updating the upstream repo multiple times? As far as I know we have not explicitly tested that use case yet, so maybe we need a separate issue for that

Reply:
Agreed. I'll add it to #2509 (creating integration tests for the git upstream functionality).